test/junit/scala/tools/nsc/reporters/ConsoleReporterTest.scala

```diff
@@ -0,0 +1,211 @@
+
+    /** checking with constructor which the same writer and echoWriter */
+    val reporter2 = createConsoleReporter("r", writerOut)
+    reporter2.echo("Hello World!")
+    testHelper(false, false, "Hello World!", "")
+  }
+
+
+  @Test
+  def printTest(): Unit = {
+    /** print test with NoPosition */
+    reporter.print(noPos, "test", reporter.INFO)
+    testHelper(false, false, "test", "")
+
+    reporter.print(noPos, "test", reporter.WARNING)
+    testHelper(false, false, "test", "warning: ")
+
+    reporter.print(noPos, "test", reporter.ERROR)
+    testHelper(false, false, "test", "error: ")
+
+    /** print test with DefinedPosition */
+    reporter.print(posWithSource, "test", reporter.INFO)
+    testHelper(false, true, "test", "")
+
+    reporter.print(posWithSource, "test", reporter.WARNING)
+    testHelper(false, true, "test", "warning: ")
+
+    reporter.print(posWithSource, "test", reporter.ERROR)
+    testHelper(false, true, "test", "error: ")
+  }
+
+
+  @Test
+  def printColumnMarkerTest(): Unit = {
+    /** printColumnMarkerTest test with NoPosition */
+    reporter.printColumnMarker(noPos)
+    testHelper(true, false, "", "")
+
+    /** printColumnMarkerTest test with DefinedPosition */
+    reporter.printColumnMarker(posWithSource)
+    assertEquals("    ^", writerOut.toString.lines.next)
+    writerOut.reset
+  }
+
+
+  @Test
+  def displayTest(): Unit = {
+    /** display test with NoPosition */
+    reporter.display(noPos, "Testing display", reporter.INFO)
+    testHelper(false, false, "Testing display", "")
+
+    reporter.display(noPos, "Testing display", reporter.WARNING)
+    testHelper(false, false, "Testing display", "warning: ")
+
+    reporter.display(noPos, "Testing display", reporter.ERROR)
+    testHelper(false, false, "Testing display", "error: ")
+
+    /** display test with DefinedPosition */
+    reporter.display(posWithSource, "Testing display", reporter.INFO)
+    testHelper(false, true, "Testing display", "")
+
+    reporter.display(posWithSource, "Testing display", reporter.WARNING)
+    testHelper(false, true, "Testing display", "warning: ")
+
+    reporter.display(posWithSource, "Testing display", reporter.ERROR)
+    testHelper(false, true, "Testing display", "error: ")
+
+    reporter.ERROR.count = 50
+    reporter.display(posWithSource, "Testing display for maxerrs to pass", reporter.ERROR)
+    testHelper(false, true, "Testing display for maxerrs to pass", "error: ")
+
+    reporter.ERROR.count = 51
+    reporter.display(posWithSource, "Testing display for maxerrs to fail", reporter.ERROR)
+    testHelper(true, false, "", "")
+
+    reporter.WARNING.count = 25
+    reporter.display(posWithSource, "Testing display for maxwarns to pass", reporter.WARNING)
+    testHelper(false, true, "Testing display for maxwarns to pass", "warning: ")
+
+    reporter.WARNING.count = 26
+    reporter.display(posWithSource, "Testing display for maxwarns to fail", reporter.WARNING)
+    testHelper(true, false, "", "")
```

Comment:
The one thing tested by `test/files/neg/maxerrs.scala` is that output is suppressed but the error or warning over the max to display is still counted, and not simply ignored. Here, the count is bumped.